components/automate-ui/src/app/pages/+compliance/+reporting/reporting-searchbar/reporting-searchbar.component.html

```diff
@@ -66,11 +66,25 @@
       <chef-icon>filter_list</chef-icon>
       <span>{{ filters.length }}</span>
     </chef-button>
-    <chef-button class="calendar-btn" secondary (click)="toggleCalendar()">
-      <chef-icon>date_range</chef-icon>
-      <span>{{ date | datetime: CHEF_SHORT_DATE }}</span>
+    <chef-button class="calendar-menu-btn" secondary (click)="toggleCalendarMenu()">
+      <span *ngIf="last24h">Last 24 hours</span>
+      <ng-container *ngIf="!last24h">
+        <chef-icon>date_range</chef-icon>
+        <span>{{ date | datetime: CHEF_SHORT_DATE }}</span>
+      </ng-container>
     </chef-button>
-    <chef-dropdown [attr.visible]="calendarVisible">
+    <chef-dropdown class="calendar-menu-dropdown" [attr.visible]="calendarMenuVisible">
+      <chef-click-outside omit="calendar-menu-btn" (clickOutside)="hideCalendarMenu()">
+        <chef-button class="select-last-btn" secondary (click)="handleSelectLast24()">
+          Last 24 hours
+        </chef-button>
+        <chef-button class="calendar-btn" secondary (click)="showCalendar()">
```

Comment:
This should not be working as written:
```suggestion
        <chef-button class="calendar-btn" secondary (click)="showCalendarMenu()">
```

Reply:
I might be misunderstanding this suggestion.

Clicking the existing datetime button in the searchbar now calls `showCalendarMenu()` which opens a new 'menu' dropdown  where you can select `last 24 hours` or select the date for a specific date in the calendar. Selecting the date option in this new menu calls the existing `showCalendar()` method to display the calendar picker as before.

Actually, it seems I didn't need to add `showCalendarMenu()` since only `toggleCalendarMenu()` and `hideCalendarMenu()` are currently being used.

Reply:
Could be my reporting error; I thought when I checked there was no `showCalendar` function.